Nadezhda Gubenko

nadezhdagub opened the pull request:
My filter is the filter of too short word

Commit:
Nadezhda Gubenko
My filter is the filter of too short word

Nekobitlz commented on src/main/java/ru/mail/polis/homework/analyzer/TooShortTextAnalyzer.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Override
+    public boolean analyze(String text) {
+        String[] word = text.split("[ ,.\"/;:)=(|!<>]+");
```
почему фильтр один и тот же вот с этим пр-ом? https://github.com/polis-vk/java-tasks/pull/1294

задумка понятное дело что очевидная, но регекс и ненужные проверки на длину одни и те же)

nadezhdagub replied:
Потому что пример данной обработки текста был взят из свободного ресурса в интернете по обработке текста на Java (не в рамках данного курса и проекта)
Убрала одну проверку на длину. Вторую оставила, поскольку проверяется именно длина каждого слова, и если хоть одно короткое встречается, то срабатывает CUSTOM (ну и взаимодействие с regex идет)

Nekobitlz replied:
Как официально) Давай проще: курс не для оценок, курс для тебя, чтобы ты научилась писать код, научилась работать с гитом и в целом получила базу для дальнейшего развития. Если ты этот код будешь откуда-то брать - мне будет пофигу, но ты код писать не научишься и это приведет к тому, что тебе дальше будет только сложнее и сложность будет расти по экспоненте. 

Тебе на работе потом не получится взять "примеры со свободных ресурсов", поэтому давай так: оставь этот код, но сделай на него норм тесты плиз

Nekobitlz commented on src/test/java/ru/mail/polis/homework/analyzer/TextFilterManagerTest.java:
```diff
@@ -141,5 +141,12 @@ private void manyFilters(boolean withPriority) {
+        TextFilterManager manager = new TextFilterManager(new TextAnalyzer[]{TextAnalyzer.createCustomAnalyzer(5)});
+        assertEquals("GOOD", manager.analyze("").toString());
+        assertEquals("GOOD", manager.analyze(null).toString());
+        assertEquals("CUSTOM", manager.analyze("Привет, я Петя :(").toString());
```
не, тестов маловато: нужны проверки на то, что если ты задаешь минимальную длину 5, то 4 и 5 пройдут, а 6 не пройдет и еще какие-нибудь кейсы придумать, например, больше кейсов со знаками препинания и комбинации русского и английского текста

nadezhdagub replied:
дополнила тесты

Nekobitlz replied:
Смотри, я хочу, чтобы ты задавала именно разную длину в параметрах у `createCustomAnalyzer`. Пусть вот эти тест кейсы останутся, но ты их прогонишь для длины 0, 5 и еще какую-нибудь возьми

Nekobitlz reviewed (commented):
По решению вопросов особо нет, это настораживает на самом деле, как-то чересчур хорошо)

Commit:
Nadezhda Gubenko
I have been doing this task for a very long time to understand what is happening and come to the current result.
Now, I removed unnecessary "public" access modifiers and renamed the method (because, in my opinion, the name was too broad in meaning) in the TextAnalyzer interface.
Completed the test of TooShortTextAnalyzer method

Nekobitlz commented on src/main/java/ru/mail/polis/homework/analyzer/TextAnalyzer.java:
```diff
@@ -8,30 +8,34 @@
+        return new TooShortTextAnalyzer(minLength);
     }
+
+    boolean analyzeFilterStatus(String text);
```
а как это исправляет ситуацию, которую я описал? название все еще не говорит о том, что происходит в методе и что он возвращает. ответь в названии на вопрос: фильтр прошел или нет?

Плюс, если название функции начинается с глагола не предполагающих какое-то возвращаемое значение (например, doSomething(), или checkSomething(), или goSomewhere() и тд), то функция должна быть void. Если хочешь назвать boolean функцию - используй глаголы по типу is, has, match (например, isNegative(), или hasMinus() и тд).

nadezhdagub replied:
Переименовала метод.

Почему-то к комментариям выше не могу написать сообщение, поэтому оставлю здесь. Тесты дополнила для разной длины

Commit:
Nadezhda Gubenko
added tests and renamed the method

Commit:
Nadezhda Gubenko
I swapped the parameters, because it makes the case more logical. Other changes in the last commit

Nekobitlz commented:
принято
